### PR TITLE
Update documentation for yeoman-generator v1.0

### PR DESCRIPTION
**Pull request by SBoudrias**
WIP. Feel free to help out by adding commits on the feature branch.

The goals are pretty simple:

1. Update API to match the changes made on the HEAD of yeoman-generator
2. Modernize the example code to es6 (subset supported by Node v4)

I'll merge that branch when I'll publish the v1 to npm.

- [x] Update to new `composeWith` signature
- [x] Update user-interaction with changes to option/argument parsing
- [x] Remove Gruntfile
- [x] `this.contextRoot`
- [x] Yarn install helpers
- [x] Update code example to ES6
- [x] Blog post announcing the Release and all changes (note: not all changes are covered in the documentation)

### CODE REVIEW

**Comment by mischah** [comment]
#701 should also be merged with the release of yeoman-generator v1.0 :octocat:

**Comment by SBoudrias** [comment]
I think that's almost complete, will just need a blog post to announce the release.

I'm hoping to release 1.0.0-rc1 over the weekend.

**Comment by mischah** [comment]
@SBoudrias Nice ✨
I thought I can help with updatading the docs. But you are just to blazing fast.

I’m not sure if I’m the right person to write that blog post. Because you have deeper knowledge.

But I could update `generator-generator` to make that available with v1 of `yeoman-genrator`.
What do you think?

**Comment by SBoudrias** [comment]
@mischah totally! Feel free to start working on it using the master branch from yeoman-generator. It's super useful to test everything works before an official release!

**Comment by SBoudrias** [comment]
Added a draft for the blog post. Let me know what you think.

**Comment by mischah** [comment]
The blog post is stellar ✨ 

What to you think about adding the PR and issue templates to the yeoman-generator repository before releasing 1.0.0 😬

**Comment by mischah** [comment]
I wonder if we should mention somewhere, that it is still possible to extend the default generator without using ES6 classes?

Or is that not necessary, because node `0.12` is almost dead.

**Comment by SBoudrias** [comment]
I added a note here https://github.com/yeoman/yeoman.github.io/pull/700/files#diff-699d56ddeed219a1042e049cd399f3ebR112

**Comment by zckrs** [comment]
I have start to update FountainJS with v1.0.0-rc1 ([PR](https://github.com/FountainJS/fountain/pull/61/files)).

Everything works well :tada: :tada: :tada:.


But I have two suggestions:
* In the `Simpler default export` part you show example for a Node version > 7. Maybe we should add example without `export` and `class` like 
  ```js
  const generators = require('yeoman-generator');
  
  module.exports = generators.extend({});
  ```

* I have strange behavior with Inquirer ([change](https://github.com/FountainJS/fountain-generator/compare/74394347cd6551fbe679bb9d40df52099d955729...55aacbfd3307ad1f1592c48b2f3d75d192c71222#diff-06f3fe995ea768fab72783b707c8157bR99)). But I will more investigate and open issue if necessary.

File: app/authoring/index.md [hunk 1]
```diff
@@ -103,32 +102,32 @@ Yeoman offers a base generator which you can extend to implement your own behavi
...
 
-module.exports = generators.Base.extend();
+module.exports = class extends Generator {};
 ```
 
-The `extend` method will extend the base class and allow you to provide a new prototype. This functionality comes from the [Class-extend](https://github.com/SBoudrias/class-extend) module and should be familiar if you've ever worked with Backbone.
-
 We assign the extended generator to `module.exports` to make it available to the ecosystem. This is how we [export modules in Node.js](https://nodejs.org/api/modules.html#modules_module_exports).
 
+If you need to support ES5 environment, the static `Generator.extend()` method can be used to extend the base class and allow you to provide a new prototype. This functionality comes from the [Class-extend](https://github.com/SBoudrias/class-extend) module and should be familiar if you've ever worked with Backbone.
```
Comment (mischah) [inline app/authoring/index.md]: @zckrs Here is the info how to extend the generator with ES5 syntax. But keep in mind that with end of life of Node `0.12` end of this month there is literally no need not to use ES6 features which came with node 4.

See: http://node.green/ for details 😘

Comment (mischah) [inline app/authoring/index.md]: @zckrs Here is the info how to extend the generator with ES5 syntax. But keep in mind that with end of life of Node `0.12` end of this month there is literally no need not to use ES6 features which came with node 4.

See: http://node.green/ for details 😘

**Comment by SBoudrias** [comment]
Released 1.0 full today! 🎉

**Comment by mischah** [comment]
Yeah 📦 🎉

**Pull request merged**